tests/benchmark/compute/instruction/test_system.py

```diff
@@ -108,6 +109,127 @@ def test_xcall(
             "during the setup phase of this test."
         )
 
+    initcode, factory_address, factory_caller_address = (
+        _deploy_max_contract_factory(pre, fork)
+    )
+
+    # Deploy num_contracts via multiple txs (each capped by tx gas limit).
+    with TestPhaseManager.setup():
+        # Rough estimate (rounded down) of contracts per tx based on dominant
+        # cost factor only. E.g., 17M gas limit + 24KiB contracts = ~3 per tx.
+        # The goal is to involve the minimum amount of gas pricing to avoid
+        # complexity and potential brittleness.
+        # If this estimation is incorrect in the future (i.e. tx gas limit cap)
+        # is increased or cost per byte, the post-state check will detect it
+        # and can be adjusted with a more complex formula.
+        num_contracts_per_tx = fork.transaction_gas_limit_cap() // (
+            gas_costs.G_CODE_DEPOSIT_BYTE * max_contract_size
+        )
+        attack_txs = math.ceil(num_contracts / num_contracts_per_tx)
+
+        contracts_deployment_txs = []
+        for _ in range(attack_txs):
+            contracts_deployment_txs.append(
+                Transaction(
+                    to=factory_caller_address,
+                    gas_limit=fork.transaction_gas_limit_cap(),
+                    gas_price=10**6,
+                    data=Hash(num_contracts_per_tx),
+                    sender=pre.fund_eoa(),
+                )
+            )
+
+    post = {}
+    for i in range(num_contracts):
+        deployed_contract_address = compute_create2_address(
+            address=factory_address,
+            salt=i,
+            initcode=initcode,
+        )
+        post[deployed_contract_address] = Account(nonce=1)
+
+    attack_call = Bytecode()
+    if opcode == Op.EXTCODECOPY:
+        attack_call = Op.EXTCODECOPY(
+            address=Op.SHA3(32 - 20 - 1, 85), dest_offset=96, size=1000
+        )
+    else:
+        # For the rest of the opcodes, we can use the same generic attack call
+        # since all only minimally need the `address` of the target.
+        attack_call = Op.POP(opcode(address=Op.SHA3(32 - 20 - 1, 85)))
```

Comment:
All this code is the same as before -- just that the diff is a bit unfortunate.

Something I noticed is that this has this `opcode == Op.EXTCODECOPY` which shouldn't be relevant for this method (look at test parametrization). Originally when I created this benchmark, the same test name was used for CALL-like opcodes and also EXTCODESIZE... but I think somebody split them after and maybe this was left here?

I'll leave this as is for now, and when I fix `test_extcode_ops` I'll try to re-use the logic since I'm pretty sure it was repeated there when it was splitted.